internal/service/auditmanager/account_registration.go

```diff
@@ -0,0 +1,193 @@
+package auditmanager
+
+import (
+	"context"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/auditmanager"
+	awstypes "github.com/aws/aws-sdk-go-v2/service/auditmanager/types"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/flex"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+func init() {
+	registerFrameworkResourceFactory(newResourceAccountRegistration)
+}
+
+func newResourceAccountRegistration(_ context.Context) (resource.ResourceWithConfigure, error) {
+	return &resourceAccountRegistration{}, nil
+}
+
+const (
+	ResNameAccountRegistration = "AccountRegistration"
+)
+
+type resourceAccountRegistration struct {
+	framework.ResourceWithConfigure
+}
+
+func (r *resourceAccountRegistration) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = "aws_auditmanager_account_registration"
+}
+
+func (r *resourceAccountRegistration) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"delegated_admin_account": schema.StringAttribute{
+				Optional: true,
+			},
+			"deregister_on_destroy": schema.BoolAttribute{
+				Optional: true,
+			},
+			"kms_key": schema.StringAttribute{
+				Optional: true,
+			},
+			"id": framework.IDAttribute(),
+			"status": schema.StringAttribute{
+				Computed: true,
+			},
+		},
+	}
+}
+
+func (r *resourceAccountRegistration) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	conn := r.Meta().AuditManagerClient
+	accountID := r.Meta().AccountID
+	region := r.Meta().Region
+	id := strings.Join([]string{accountID, region}, ",")
+
+	var plan resourceAccountRegistrationData
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	in := auditmanager.RegisterAccountInput{}
+	if !plan.DelegatedAdminAccount.IsNull() {
+		in.DelegatedAdminAccount = aws.String(plan.DelegatedAdminAccount.ValueString())
+	}
+	if !plan.KmsKey.IsNull() {
+		in.KmsKey = aws.String(plan.KmsKey.ValueString())
+	}
+	out, err := conn.RegisterAccount(ctx, &in)
```

Comment:
What happens if `RegisterAccount` is called when the account is already registered?
Given the `deregister_on_destroy` flag there is a possibility that this will happen even if everything is managed by Terraform.

Reply:
This functions like a typical `Put*` API. Calling it when the account is already active just sets the KMS key and delegated account ID values to the new input and keeps the status as `ACTIVE`.